### ISSUE TITLE
Test failure when building the PICT plugin on Java 8

### ISSUE THREAD

**Issue by octavian-nita** [haraldk/TwelveMonkeys#94]
Hi,

I've just checked out the latest version of TwelveMonkeys and tried to build it from the main pom file. However, the build (tests actually) fails at the PICT plugin.

Test machine / configuration:
 * Windows 7 Pro (Mac OS X fails as well)
 * jdk8
 * <sun.java2d.cmm>sun.java2d.cmm.kcms.KcmsServiceProvider</sun.java2d.cmm> set in the main pom.xml file

Below, the individual test report:
\-------------------------------------------------------------------------------
Test set: com.twelvemonkeys.imageio.plugins.pict.PICTImageWriterTest
\-------------------------------------------------------------------------------
Tests run: 17, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.221 sec <<< FAILURE!
testWriteReadCompare(com.twelvemonkeys.imageio.plugins.pict.PICTImageWriterTest)  Time elapsed: 0.145 sec  <<< FAILURE!
java.lang.AssertionError: Test data 6 R(10,0) expected:<851968.0> but was:<393216.0>
	at org.junit.Assert.fail(Assert.java:91)
	at org.junit.Assert.failNotEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:441)
	at com.twelvemonkeys.imageio.plugins.pict.PICTImageWriterTest.testWriteReadCompare(Unknown Source)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:76)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
	at org.apache.maven.surefire.junit4.JUnit4TestSet.execute(JUnit4TestSet.java:53)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:123)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:104)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:164)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:110)
	at org.apache.maven.surefire.booter.SurefireStarter.invokeProvider(SurefireStarter.java:175)
	at org.apache.maven.surefire.booter.SurefireStarter.runSuitesInProcessWhenForked(SurefireStarter.java:107)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:68)

**Comment by haraldk** [comment]
Hi Octavian,

Thanks for reporting! This is a semi-known issue. The build (the tests) doesn't fully work on OpenJDK/Java 8 (read: using LCMS) yet. I want to fix this for 3.1, and it's one of the reasons I haven't released it yet. I'm working on it (most of it was fixed for #41). :-)

Anyways, setting `-Dsun.java2d.cmm=sun.java2d.cmm.kcms.KcmsServiceProvider` for the tests works for me (on OS X, Java 1.8.0_25). I think the problem is that you are not setting it at the right place. The setting must be passed to the surefire-plugin like this, because each test run is forked to its own VM:

                <plugin>
                    <groupId>org.apache.maven.plugins</groupId>
                    <artifactId>maven-surefire-plugin</artifactId>
                    <version>2.10</version>
                    <configuration>
                        <systemProperties>
                            <property>
                                <name>java.awt.headless</name>
                                <value>true</value>
                            </property>
                            <property>
                                <name>sun.java2d.cmm</name>
                                <value>sun.java2d.cmm.kcms.KcmsServiceProvider</value>
                            </property>
                        </systemProperties>
                    </configuration>
                </plugin>

Hope this can serve as a workaround for now, until the issue is fully fixed. :-)


Best regards,

-- 
Harald K

**Comment by octavian-nita** [comment]
Hi Harald,

Thanks a lot -- indeed, setting the property for the Surefire plugin instead of doing it "globally" did the trick, at least on Windows (no reason why it shouldn't on my Mac also, but I haven't tried yet). Apparently Surefire forks a new jvm (at least by default) so we kinda' need to set that one up with the property.

P.S. one can also set arguments for Surefire using the `argLine` parameter (apparently sometimes this is the only way to make some properties visible, but it is not the case for `sun.java2d.cmm`):

```
<configuration>
  <argLine>-Djava.security.manager -Dsun.java2d.cmm=sun.java2d.cmm.kcms.KcmsServiceProvider\</argLine>
...
```
